src/Features/CSharp/Portable/CodeFixes/FixReturnType/CSharpFixReturnTypeCodeFixProvider.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.FixReturnType
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.FixReturnType), Shared]
+    internal class CSharpFixReturnTypeCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        // error CS0127: Since 'M()' returns void, a return keyword must not be followed by an object expression
+        private const string CS0127 = nameof(CS0127);
+
+        // error CS1997: Since 'M()' is an async method that returns 'Task', a return keyword must not be followed by an object expression. Did you intend to return 'Task<T>'?
+        private const string CS1997 = nameof(CS1997);
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CS0127, CS1997);
+
+        public CSharpFixReturnTypeCodeFixProvider()
+            : base(supportsFixAll: false)
+        {
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            if (await TryGetOldAndNewReturnTypeAsync(context.Document, context.Diagnostics, context.CancellationToken).ConfigureAwait(false) == default)
```

Comment:
1. so long.  and hard to see the `== default`.  can you break into separate statements?
2. consider locals for 'document/diagnostics'.